design/task_model.md

```diff
@@ -132,9 +132,15 @@ way to `RUNNING`. If a task exits without an error code, it moves to the
 `COMPLETE` state. If it fails, it moves to the `FAILED` state instead.
 
 A task may alternatively end up in the `SHUTDOWN` state if its shutdown was
-requested by the orchestrator, the `REJECTED` state if the agent rejected the
+requested by the orchestrator (by setting desired state to `SHUTDOWN`),
+the `REJECTED` state if the agent rejected the
 task, or the `ORPHANED` state if the node on which the task is scheduled is
-down for too long.
+down for too long. The orchestrator might also set desired state for a task to
+`REMOVE` when the service associated with the task was removed or scaled down
+by the user. When this happens, the agent proceeds to shut the task down.
+The task is only fully removed by the task reaper when the shutdown succeeds.
+Notice that tasks that were removed becacuse of service removal or scale down
```

Comment:
This line is probably redundant with the change above.

Reply:
Redundant because of the line "The task is removed from the store by the task reaper only after the shutdown succeeds." 

Removed from the store == not kept around in task history.

But its not a big deal.

Reply:
Gotcha. Still think it is useful to state explicitly for clarity.